src/server/frontend_wayland/output_manager.cpp

```diff
@@ -213,19 +215,42 @@ void mf::Output::resource_destructor(wl_resource* resource)
     client_resource_list.erase(erase_from, client_resource_list.end());
 }
 
+struct mf::OutputManager::DisplayConfigObserver: public graphics::NullDisplayConfigurationObserver
+{
+    DisplayConfigObserver(OutputManager& manager)
+        : manager{manager}
+    {
+    }
+
+private:
+    OutputManager& manager;
+
+    void initial_configuration(std::shared_ptr<graphics::DisplayConfiguration const> const& config) override
+    {
+        manager.handle_configuration_change(config);
+    }
+
+    void configuration_applied(std::shared_ptr<graphics::DisplayConfiguration const> const& config) override
+    {
+        manager.handle_configuration_change(config);
+    }
+};
 
-mf::OutputManager::OutputManager(wl_display* display, std::shared_ptr<MirDisplay> const& display_config, std::shared_ptr<Executor> const& executor) :
-    display_config_{display_config},
-    display{display},
-    executor{executor}
+mf::OutputManager::OutputManager(
+    wl_display* display,
+    std::shared_ptr<Executor> const& executor,
+    std::shared_ptr<ObserverRegistrar<graphics::DisplayConfigurationObserver>> const& registrar)
+    : display{display},
+      executor{executor},
```

Comment:
You should probably store the executor in the `DisplayConfigObserver` rather than the `OutputManager`; then you maintain the invariant `ObserverRegistrar` wants which is that the Executor lives at least as long as the Observer.